flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1612,10 +1689,10 @@ def assert_calculated_changes(
 
     :param discovered_datasets: Collection of ``DiscoveredDataset`` to
         expose as local state.
+    :param api: ``IBlockDeviceAPI`` provider to use; by default a
```

Comment:
I think I assumed that assertion did discovery too, but it doesn't. calculate_changes should be functional, yeah.